src/Auth.ts

```diff
@@ -750,6 +837,63 @@ export class Auth {
       return resource;
     }
   }
+
+  private async getAllConnectionsFromStorage(): Promise<Connection[]> {
+    const connectionStorage = this.getAllConnectionsStorage();
+    const json: string = await connectionStorage.get();
+    return JSON.parse(json);
+  }
+
+  public async switchToConnection(connection: Connection): Promise<void> {
+    this.connection.deactivate();
+    this._connection = Object.assign(this._connection, connection);
+    this._connection.active = true;
+    await this.storeConnectionInfo();
+  }
+
+  public async updateConnection(oldName: string, newName: string): Promise<void> {
+    const allConnections = await this.getAllConnections();
+
+    if (allConnections.filter(c => c.name === newName).length > 0) {
+      throw new CommandError(`The connection name '${newName}' is already in use`);
+    }
+
+    allConnections.forEach(c => {
```

Comment:
To avoid iterating over all connections twice, let's update the code, so that we store the outcome of allConnections.filter in an array. We can then use it to compare its length first, and if we found only one object, directly update its name.

Reply:
I've changed it a bit, as there's only one entry of the connection available.